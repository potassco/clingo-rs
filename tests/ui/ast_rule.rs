use clingo::ast::*;
use clingo::{Location, Symbol};

fn main() {
    let loc = Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term = symbolic_term(&loc, &sym).unwrap();
    let atom = symbolic_atom(term).unwrap();
    let lit = basic_literal_from_symbolic_atom(&loc, Sign::NoSign, atom).unwrap();
    let hlit: Head = lit.clone().into();
    let blit: BodyLiteral = lit.into();
    let body = vec![blit];
    let rule = rule(&loc, hlit, &body);
    drop(body);
    let _end = rule;
}
