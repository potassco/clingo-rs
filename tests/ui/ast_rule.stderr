error[E0505]: cannot move out of `body` because it is borrowed
  --> $DIR/ast_rule.rs:14:10
   |
13 |     let rule = rule(&loc, hlit, &body);
   |                                 ----- borrow of `body` occurs here
14 |     drop(body);
   |          ^^^^ move out of `body` occurs here
15 |     let _end = rule;
   |                ---- borrow later used here
